### ISSUE TITLE
Support configurable save names and locations in Console

### ISSUE THREAD

**Issue by Extender92** [Extender92/Monopoly#16]
## Problem/Goal

Console currently lacks a clear way to choose a save name and configured save location. Add frontend-owned save targeting without leaking file paths into Core or disrupting the active turn/session.

## Required behavior

- Console lets the user choose a valid save name and a configured storage location exposed by the injected save store.
- Existing targets require an explicit overwrite confirmation.
- Cancellation returns to the current session without changing match state.
- Storage, permission, invalid-name, and atomic-write failures are presented as clear frontend errors.
- Infrastructure performs the atomic write; Core never reads Console input or handles file-system paths.
- Saving neither advances turn state nor recursively opens another menu/session loop.

## Scope

- Console save-name/location prompts and validation feedback.
- Overwrite/cancel navigation results.
- Wiring to the injected persistence abstraction and frontend tests.

## Out of scope

- Serializer/schema design, cloud storage, or database storage.
- Loading and save-list selection, owned by #17.
- Session composition, owned by #18.
- Game-rule changes.

## Acceptance criteria

- A user can save to an allowed configured location with a chosen valid name.
- Existing targets are never replaced without explicit confirmation.
- Cancel, invalid input, and storage failures leave the active match and current turn unchanged.
- Console shows actionable errors without exposing an unhandled exception.
- Core has no file paths, storage prompts, or Console dependencies.
- Save completion returns through the explicit session loop without recursive menu calls.
- Fake-store Console tests cover success, overwrite accept/decline, cancel, invalid name, and storage failure.
- Full Release build and test suite pass without new warnings.

## Dependencies

- Blocked by shared session composition #18 and Infrastructure extraction #33.
- Consume the supported save format from #52 once Version 2 lands.

## Related issues

- Related to #17
- Related to #20

**Comment by Extender92** [comment]
Deferred during neutral-Core backlog sanitation. This issue has not been implemented. It will not be copied to the clean repository and is not a cutover prerequisite. Reconsider the behavior only as a new, independently scoped neutral issue after cutover, based on the clean architecture and current product goals.
